src/components/poolstats/PoolPositionWidget.tsx

```diff
@@ -39,10 +49,175 @@ const PerformanceCard = styled.div`
+}
+
+function findNearestElementByTime<T extends Timestamp>(arr: T[], timestamp: string | number): T {
+  const costFunc = arr.map((el) => Math.abs(Number(el.timestamp) - Number(timestamp)));
+  const idx = costFunc.reduce((iMin, cost, i, arr) => cost < arr[iMin] ? i : iMin, 0);
+  return arr[idx];
+}
+
 export default function PoolPositionWidget(props: PoolPositionWidgetProps) {
-  // TODO: Incorporate the pool data into the widget
+
+  const { poolData, offChainPoolStats } = props;
+  const { poolStats } = useContext(BlendPoolContext);
+
+  const [{ data: accountData }] = useAccount();
+  const [{ data: accountShareBalance }] = useBalance({
+    addressOrName: accountData?.address,
+    token: poolData.poolAddress,
+    watch: false,
+  });
+
+  const [accountStats, setAccountStats] = useState<AccountStats | null>(null)
+
+  useEffect(() => {
+    let mounted = true;
+
+    if (offChainPoolStats && poolStats && accountShareBalance && accountData) {
+      const tvl = new Big(offChainPoolStats.total_value_locked);
+      const accountValue = tvl.mul(toBig(accountShareBalance.value)).div(poolStats.outstandingShares);
+      const accountAddress = accountData.address.slice(2).toLowerCase();
+
+      const getEvents = makeEtherscanRequest(
+        BLEND_FACTORY_CREATION_BLOCK,
+        poolData.poolAddress,
+        [
+          null,
+          `0x000000000000000000000000${accountAddress}`, // to this user
+          `0x000000000000000000000000${accountAddress}` // from this user
+        ],
+        false // `or` between topics, not `and`
+      );
+
+      const endTime = Date.now() / 1000;
+      // Get all-time data for "Total Returns"
+      const getPoolReturns = axios.get(`${API_URL}/pool_returns/${poolData.poolAddress}/1/all/${endTime.toFixed(0)}`);
+      const getPrices0 = axios.get(`${API_URL}/token_returns/${poolData.token0Address}/1/all/${endTime.toFixed(0)}`);
+      const getPrices1 = axios.get(`${API_URL}/token_returns/${poolData.token1Address}/1/all/${endTime.toFixed(0)}`);
+      // Get 3-month data for "Today's Returns" (we need data from exactly 24 hours ago, and 3m gives us that without as many extraneous entries as 1d)
+      const getPoolReturns3M = axios.get(`${API_URL}/pool_returns/${poolData.poolAddress}/1/3m/${endTime.toFixed(0)}`);
+      const getPrices03M = axios.get(`${API_URL}/token_returns/${poolData.token0Address}/1/3m/${endTime.toFixed(0)}`);
+      const getPrices13M = axios.get(`${API_URL}/token_returns/${poolData.token1Address}/1/3m/${endTime.toFixed(0)}`);
+
+      axios.all([getEvents, getPoolReturns, getPrices0, getPrices1, getPoolReturns3M, getPrices03M, getPrices13M]).then(
+        axios.spread((eventData, poolReturnsData, price0Data, price1Data, poolReturnsData3M, price0Data3M, price1Data3M) => {
+          const events = eventData.data.result;
+          const transfers = events.filter((ev: any) => ev.topics[0] === '0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef');
```

Comment:
Should we utilize constants here?

Reply:
We may implement similar event checking for the portfolio page, in which case we can generalize this. But for now it's a one-off thing